src/applications/mhv/secure-messaging/tests/components/MessageListComponents/folderHeader.unit.spec.jsx

```diff
@@ -30,41 +38,275 @@ describe('FolderHeader component in custom folder', () => {
       },
     },
   };
-  const setup = (state = initialState) => {
+  const initialPath = `/folders/${customFolder.folderId}`;
+  const setup = (
+    state = initialState,
+    path = initialPath,
+    threadCount = threadList,
+    folder = customFolder,
+  ) => {
     return renderWithStoreAndRouter(
       <FolderHeader
-        folder={{
-          folderId: 7038175,
-          name: 'TEST2',
-          count: 1,
-          unreadCount: 0,
-          systemFolder: false,
-        }}
+        folder={folder}
+        threadCount={threadCount}
         searchProps={{ ...searchProps }}
       />,
       {
         initialState: state,
         reducers: reducer,
-        path: '/folders/7038175',
+        path,
       },
     );
   };
 
-  let screen = null;
-  beforeEach(() => {
-    screen = setup();
+  describe('Folder Header component displays CUSTOM folder and children components', () => {
+    it('must display valid CUSTOM FOLDER name and description: DEMO FOLDER 1', async () => {
+      const screen = setup();
+      expect(screen.getByText(customFolder.name)).to.exist;
+      expect(screen.getByText(Folders.CUSTOM_FOLDER.desc)).to.exist;
+    });
+
+    it('renders FilterBox with `threadCount` in CUSTOM FOLDER', () => {
+      const screen = setup();
+      const filterBox = screen.getByTestId('search-form');
+      expect(filterBox).to.exist;
+
+      const filterHeader = filterBox.querySelector('h2');
+      expect(filterHeader.textContent).to.equal(
+        `Filter messages in ${customFolder.name} `,
+      );
+
+      const filterInputElement = screen.queryByTestId('keyword-search-input');
+      expect(filterInputElement.getAttribute('label')).to.equal(
+        'Enter information from one of these fields: to, from, message ID, or subject',
+      );
+      expect(filterInputElement.getAttribute('aria-label')).to.equal(
+        `Filter messages in ${
+          customFolder.name
+        } Enter information from one of these fields: to, from, message ID, or subject`,
+      );
+    });
+
+    it('does not render FilterBox w/o `threadCount` on CUSTOM FOLDER', () => {
+      const screen = setup(initialState, initialPath, null);
+      expect(screen.queryByTestId('search-form')).to.not.exist;
+    });
   });
 
-  it('must display valid custom folder name', async () => {
-    expect(screen.getByText('TEST2')).to.exist;
-    expect(screen.getByText(Folders.CUSTOM_FOLDER.desc)).to.exist;
+  describe('Folder Header component displays INBOX folder and children components', () => {
+    const initialInboxState = {
+      sm: {
+        folders: {
+          folder: inbox,
+          folderList,
+        },
+      },
+      user: {
+        profile: {
+          session: {
+            ssoe: true,
+          },
+        },

```

Comment:
Just curious what is this user reducer required for?

Reply:
Not really used for the tests.. was originally using it to see it to see if it would give the user an `authorized` status instead of `not authorized`.